lib/src/tabs/Tabs.tsx

```diff
@@ -1,13 +1,43 @@
-// @ts-nocheck
-import React, { useState } from "react";
-import Tabs from "@material-ui/core/Tabs";
-import Tab from "@material-ui/core/Tab";
+import React, { useCallback, useEffect, useMemo, useRef, useState } from "react";
 import styled, { ThemeProvider } from "styled-components";
-
 import { spaces } from "../common/variables.js";
-import DxcBadge from "../badge/Badge";
 import useTheme from "../useTheme";
-import TabsPropsType from "./types";
+import useTranslatedLabels from "../useTranslatedLabels";
+import Tab from "./Tab";
+import TabsPropsType, { Margin, Space } from "./types";
+
+const arrowIcons = {
+  left: (
+    <svg focusable="false" viewBox="0 0 24 24" aria-hidden="true" role="img">
+      <path d="M15.41 16.09l-4.58-4.59 4.58-4.59L14 5.5l-6 6 6 6z"></path>
+    </svg>
+  ),
+  right: (
+    <svg focusable="false" viewBox="0 0 24 24" aria-hidden="true" role="img">
+      <path d="M8.59 16.34l4.58-4.59-4.58-4.59L10 5.75l6 6-6 6z"></path>
+    </svg>
+  ),
+};
+
+const useResize = (refTabList) => {
+  const [viewWidth, setViewWidth] = useState(0);
+
+  const handleWindowSizeChange = useCallback(() => {
+    setViewWidth(refTabList?.current?.offsetWidth);
+  }, [refTabList]);
+
+  useEffect(() => {
+    window.addEventListener("load", handleWindowSizeChange);
+    window.addEventListener("resize", handleWindowSizeChange);
+
+    return () => {
+      window.removeEventListener("load", handleWindowSizeChange);
+      window.removeEventListener("resize", handleWindowSizeChange);
+    };
+  }, [refTabList, handleWindowSizeChange]);
+
```

Comment:
Are `refTabList` and `handleWindowSizeChange` really needed as dependencies in useEffect?

Reply:
Removed refTabList dependency, left the other one just in case.